test/test_query.cpp

```diff
@@ -5651,13 +5651,15 @@ TEST(Query_NestedLinkCount)
     o3.get_linklist("children").add(o2.get_key());
 
     auto q = table->query("children.list.@size == 0");
-    CHECK_EQUAL(q.count(), 2);
+    CHECK_EQUAL(q.count(), 3); // all lists are empty
```

Comment:
As stated above, I will argue that o1 does not even have a list - hence we cannot say anything about the size.

Reply:
I agree with Jørgen here - it is quite counterintuitive to create a query like "give me all TestClass-es which have at least one element in `children` that has an empty `list`" and then get an object that has no elements in `children`. The base requirement to evaluate would be to have "at least one element in `children`".

Reply:
That makes sense to me; I have been persuaded by this argument. I'd just request that SDKs make a note about this change in behaviour in their release notes in case apps were relying on the previous results.